### DISCUSSION TITLE
Welcome to csharp Discussions!

### DISCUSSION THREAD

**Discussion by tg123** [kubernetes-client/csharp#858] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lazarshulekov** [comment]
I had a look at the library source code and it seems the lib tries to acquire the lease and once done it calls OnStartedLeading?.Invoke();
After that it starts a lease renew loop.The problem we have is when the OnStartedLeader handler takes 30 seconds but the lease TTL is only 5 seconds, meaning that the current leader will block on OnStartedLeader, thinking it is the current leader but since the lease ttl is only 5 seconds, there will be another instance already becoming the leader, so we could have two parallel leaders at the same time.
This could be solved by increasing the TTL of the lease to a value bigger than the expected OnStartedLeading but since we don't have a way to forcefully release the lease, we are going to wait for the TTL each time the apps switches the leader.
